tools/c7n_org/c7n_org/cli.py

```diff
@@ -199,28 +199,31 @@ def resolve_regions(regions):
 
 
 def get_session(account, session_name, region):
-    if account.get('role'):
-        roles = account['role']
-        if isinstance(roles, str):
-            roles = [roles]
-        s = None
-        for r in roles:
-            try:
-                s = assumed_session(
-                    r, session_name, region=region,
-                    external_id=account.get('external_id'),
-                    session=s)
-            except ClientError as e:
-                log.error(
-                    "unable to obtain credentials for account:%s role:%s error:%s",
-                    account['name'], r, e)
-                raise
-        return s
-    elif account.get('profile'):
-        return SessionFactory(region, account['profile'])()
+    if account.get('provider') == 'aws':
+        if account.get('role'):
```

Comment:
rather than nesting these two, can we just do a compound boolean ie. `account.get() == 'aws' and account.get()`

Reply:
Instead of nesting perhaps just have an early exit? Then the rest of this block doesn't need to get indented.

```python
if account.get('provider') != 'aws':
    return None
# Then the rest of the function as it was
```

Reply:
code updated following the @howbazaar reco